cdap-docs/developers-manual/source/advanced/adapters.rst

```diff
@@ -0,0 +1,140 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+================
+Adapters
+================
+
+Adapters read data from one or more data sources, perform a transformation, then
+write to one or more data sinks. They may be used for ETL purposes, or any situation
+where you want to pipe data from one data source to another.
+
+CDAP currently provides a stream-conversion adapter that regularly reads data from
+a Stream, then writes the data to a TimePartitionedFileSet in avro format. The output
+can then be explored through Hive or Impala, assuming a compatible schema is used. 
+
+Each Adapter type is implemented as a CDAP Application, with each Adapter instance as
+a schedule for running a program in the application. For example, the stream-conversion
+Adapter type is an application named StreamConversion. Each Adapter instance is a schedule
+that periodically runs the Workflow defined in the StreamConversion application.
+Custom adapters can be added to CDAP, though the feature is experimental and will
+likely change and improve in future releases.
+
+Stream Conversion Adapter
+=========================
+The stream-conversion Adapter type regularly reads data from a source Stream and writes
+it to a TimePartitionedFileSet as avro files. You may want to use this Adapter if you
```

Comment:
``TimePartitionedFileSet`` as Avro files. \[Avro should be capitalized. Can we add a Glossary entry for Avro? Just open up the reference-manual/glossary.rst, and add it in. Goes at the bottom, as Sphinx sorts and builds the glossary...]